apps/expo-go/android/expoview/src/main/java/host/exp/exponent/modules/perfmonitor/PerfMonitorDataSource.kt

```diff
@@ -0,0 +1,260 @@
+package host.exp.exponent.modules.perfmonitor
+
+import android.os.Handler
+import android.os.Looper
+import android.view.Choreographer
+import androidx.annotation.AnyThread
+import androidx.annotation.MainThread
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactMarker
+import com.facebook.react.bridge.UiThreadUtil
+import com.facebook.react.fabric.DevToolsReactPerfLogger
+import com.facebook.react.fabric.FabricUIManager
+import com.facebook.react.uimanager.UIManagerHelper
+import com.facebook.react.uimanager.common.UIManagerType
+import kotlin.math.max
+import kotlin.math.min
```

Comment:
```suggestion
class WeakHolder<T> {
  private var current = WeakReference<T>(null)

  operator fun getValue(thisRef: Any?, property: KProperty<*>): T? {
    return current.get()
  }

  operator fun setValue(thisRef: Any?, property: KProperty<*>, value: T?) {
    current = WeakReference<T>(value)
  }
}
```
I think adding something like this would be helpful. We can maybe add it to the core 🤔